sidechain/test/src/sidechain_block_builder.rs

```diff
@@ -0,0 +1,139 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! Builder pattern for a signed sidechain block.
+
+use itp_time_utils;
+use itp_types::H256;
+use its_primitives::{
+	traits::{Block as BlockTrait, SignBlock},
+	types::{
+		block::{BlockHash, BlockNumber, Timestamp},
+		Block, ShardIdentifier, SignedBlock,
+	},
+};
+use sp_core::{ed25519, Pair};
+
+type Seed = [u8; 32];
+const ENCLAVE_SEED: Seed = *b"12345678901234567890123456789012";
+
+pub struct SidechainBlockBuilder {
+	signer: ed25519::Pair,
+	number: BlockNumber,
+	parent_hash: BlockHash,
+	parentchain_block_hash: H256,
+	signed_top_hashes: Vec<H256>,
+	encrypted_payload: Vec<u8>,
+	shard: ShardIdentifier,
+	timestamp: Timestamp,
+}
+
+impl Default for SidechainBlockBuilder {
+	fn default() -> Self {
+		SidechainBlockBuilder {
+			signer: Pair::from_seed(&ENCLAVE_SEED),
+			number: 1,
+			parent_hash: BlockHash::default(),
+			parentchain_block_hash: Default::default(),
+			signed_top_hashes: Default::default(),
+			encrypted_payload: Default::default(),
+			shard: Default::default(),
+			timestamp: Default::default(),
+		}
+	}
+}
+
+impl SidechainBlockBuilder {
+	pub fn random() -> Self {
+		SidechainBlockBuilder {
+			signer: Pair::from_seed(&ENCLAVE_SEED),
+			number: 42,
+			parent_hash: BlockHash::random(),
+			parentchain_block_hash: BlockHash::random(),
+			signed_top_hashes: vec![H256::random(), H256::random()],
+			encrypted_payload: vec![1, 3, 42, 8, 11, 33],
+			shard: ShardIdentifier::random(),
+			timestamp: itp_time_utils::now_as_u64(),
+		}
+	}
```

Comment:
It can be nice to have this `random()` method - from my experience though it can make tests flaky and very very difficult to debug (when they fail). We can keep it for now, since it shouldn't be a problem (at least I hope not). But in general, if there are values that you think are problematic, write a specific test for them. And have tests for 'regular', but deterministic values.

Reply:
Ok, that's good to know. I was thinking about that as well, but I also thought that a random function could be useful. I'll remember that the next time.